### DISCUSSION TITLE
Model binding with a primary key containing the table name doesn't work correctly

### DISCUSSION THREAD

**Discussion by royduin** [livewire/livewire#3031] [7. Bugs (archived)]
### Description

I'm getting an error when Livewire is making requests with changes:
```
Missing required parameter for [Route: issue] [URI: issue/{issue}] [Missing parameter: issue]
```
And when I look into this it's caused by the primary key on the model which includes the table name:
```php
protected $primaryKey = 'issues.id';
```
When I remove the scopes which require this as the joined tables also include an `id` column and remove this custom primary key it's working.

### Exact steps to reproduce

Create a model with the table name in the primary key, for example:
```php
class Issue extends Model
{
    protected $primaryKey = 'issues.id';
}
```

And use that model in a component:
```php
class Issue extends Component
{
    public App\Models\Issue $issue;
    public $something = null;

    public function render()
    {
        return view('livewire.issue');
    }
}
```

And bind something to `$something` in the view:
```html
Issue: {{ $issue->id }}
<input type="text" wire:model="something"/>
```

Now type something in the input.

### Stripped-down, copy-pastable code snippets

**Context**
- Livewire version: 2.4.2
- Laravel version: 8.36.2

**Comment by ph7jack** [comment]
This is not a livewire bug, i created a pr to solves it on laravel

**Comment by driesvints** [comment]
The actual issue here is that you shouldn't define the table on the primary key. I don't know why you're doing that?

**Comment by GrahamCampbell** [comment]
There is no bug. Your primary key value is invalid. You must write:

```php
protected $primaryKey = 'id';
```

**Comment by royduin** [comment]
So if you want to join a table (from for example a (global) scope) and both tables have an id, what should you use then?
```php
Issue::find(1)
``` 
Will cause issues because both tables have an id. I could write that more explicit with for example:
```php
Issue::firstWhere('issues.id', 1)
``` 
But implicit route model binding won't work. That's why I added the table name to `$primaryKey`

**Comment by driesvints** [comment]
The first example should never clash with a global scope issue. Most likely you're applying joins in a global scope that you shouldn't be doing in the first place. That'll indeed cause the issues you're experiencing.

**Comment by royduin** [comment]
Using multiple subselects on the same table for now, less efficient in comparison with a join but no conflicts.
